judge/views/api/api_v2.py

```diff
@@ -1,122 +1,579 @@
+        can_see_rankings = contest.can_see_scoreboard(self.request.user)
+        if contest.hide_scoreboard and in_contest:
+            can_see_rankings = False
+        can_see_problems = (in_contest or contest.ended or contest.is_editable_by(self.request.user))
+
+        problems = list(
+            contest.contest_problems
+            .select_related('problem')
+            .defer('problem__description')
+            .order_by('order'),
+        )
+
+        new_ratings_subquery = Rating.objects.filter(participation=OuterRef('pk'))
+        old_ratings_subquery = (
+            Rating.objects
+            .filter(user=OuterRef('user__pk'), contest__end_time__lt=OuterRef('contest__end_time'))
+            .order_by('-contest__end_time')
+        )
+        participations = (
+            contest.users
+            .filter(virtual=ContestParticipation.LIVE, user__is_unlisted=False)
+            .annotate(
+                username=F('user__user__username'),
+                old_rating=Subquery(old_ratings_subquery.values('rating')[:1]),
+                new_rating=Subquery(new_ratings_subquery.values('rating')[:1]),
+            )
+            .order_by('-score', 'cumtime')
+        )
+
+        return {
+            'key': contest.key,
+            'name': contest.name,
+            'start_time': contest.start_time.isoformat(),
+            'end_time': contest.end_time.isoformat(),
+            'time_limit': contest.time_limit and contest.time_limit.total_seconds(),
+            'is_rated': contest.is_rated,
+            'rate_all': contest.is_rated and contest.rate_all,
+            'has_rating': contest.ratings.exists(),
+            'rating_floor': contest.rating_floor,
+            'rating_ceiling': contest.rating_ceiling,
+            'is_organization_private': contest.is_organization_private,
+            'organizations': list(contest.organizations.values_list('id', flat=True)),
+            'is_private': contest.is_private,
+            'tags': list(contest.tags.values_list('name', flat=True)),
+            'format': {
+                'name': contest.format_name,
+                'config': contest.format_config,
+                'problem_label_script': contest.problem_label_script,
+            },
+            'problems': [
+                {
+                    'points': int(problem.points),
+                    'partial': problem.partial,
+                    'is_pretested': problem.is_pretested and contest.run_pretests_only,
+                    'max_submissions': problem.max_submissions or None,
+                    'name': problem.problem.name,
+                    'code': problem.problem.code,
+                } for problem in problems
+            ] if can_see_problems else [],
+            'rankings': [
+                {
+                    'user': participation.username,
+                    'score': participation.score,
+                    'cumtime': participation.cumtime,
```

Comment:
Might as well take the opportunity here to give this a better name than `cumtime`, so that it matches the rest.

Reply:
This has been fixed.